ldi-nifi/ldi-nifi-processors/ldes-client-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/LdesClient.java

```diff
@@ -0,0 +1,138 @@
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Set;
+
+import static be.vlaanderen.informatievlaanderen.ldes.ldi.config.LdesProcessorProperties.*;
+import static be.vlaanderen.informatievlaanderen.ldes.ldi.config.LdesProcessorRelationships.DATA_RELATIONSHIP;
+
+@SuppressWarnings("java:S2160") // nifi handles equals/hashcode of processors
+@Tags({ "ldes-client", "vsds" })
+@CapabilityDescription("Extract members from an LDES source and send them to the next processor")
+@Stateful(description = "Stores mutable fragments to allow processor restart", scopes = Scope.LOCAL)
+public class LdesClient extends AbstractProcessor {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(LdesClient.class);
+
+	protected String identifier = null;
+	protected LdesClientConfig config = new LdesClientConfig();
+	protected LdesService ldesService;
+	private final EndpointRequester endpointRequester = new EndpointRequester();
+
+	@Override
+	public Set<Relationship> getRelationships() {
+		return Set.of(DATA_RELATIONSHIP);
+	}
+
+	@Override
+	public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+		return List.of(DATA_SOURCE_URL, DATA_SOURCE_FORMAT, DATA_DESTINATION_FORMAT, FRAGMENT_EXPIRATION_INTERVAL,
+				STREAM_TIMESTAMP_PATH_PROPERTY, STREAM_VERSION_OF_PROPERTY, STREAM_SHAPE_PROPERTY);
+	}
+
+	@OnAdded
+	public void onAdded() {
+		if (identifier == null) {
+			identifier = getIdentifier();
+		}
+		config.setPersistenceDbName(identifier + "-" + config.getPersistenceDbName());
+	}
+
+	@OnScheduled
+	public void onScheduled(final ProcessContext context) {
+		String dataSourceUrl = LdesProcessorProperties.getDataSourceUrl(context);
+		Lang dataSourceFormat = LdesProcessorProperties.getDataSourceFormat(context);
+		Long fragmentExpirationInterval = LdesProcessorProperties.getFragmentExpirationInterval(context);
+
+		ldesService = LdesClientImplFactory.getLdesService(config);
+
+		ldesService.setDataSourceFormat(dataSourceFormat);
+		ldesService.setFragmentExpirationInterval(fragmentExpirationInterval);
+		ldesService.queueFragment(getStartingUrl(dataSourceUrl, dataSourceFormat));
+
+		LOGGER.info("LDES extraction processor {} with base url {} (expected LDES source format: {})",
+				context.getName(), dataSourceUrl, dataSourceFormat);
+	}
+
+	private String getStartingUrl(String dataSourceUrl, Lang dataSourceFormat) {
+		return endpointRequester
+				.determineStartingNode(new Endpoint(dataSourceUrl, dataSourceFormat))
+				.map(StartingNode::url)
+				.orElseThrow(
+						() -> new IllegalArgumentException(
+								"Starting url could not be determined for fragmentId: " + dataSourceUrl));
+	}
+
+	@Override
+	public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
```

Comment:
we should decrease the complexity of this method

Reply:
Out of scope of this PR, but to be picked up in dedicated dev session